### ISSUE TITLE
type '() => Null' is not a subtype of type '(() => StoryItem)?' of 'orElse'

### ISSUE THREAD

**Issue by blacknight9** [awaik/flutter_instagram_stories#20]
when opening the story i get this error

`type '() => Null' is not a subtype of type '(() => StoryItem)?' of 'orElse'
`
can you help

**Comment by awaik** [comment]
Probably you have the wrong name in one of the fields in the database.

**Comment by blacknight9** [comment]
> Probably you have the wrong name in one of the fields in the database.

here is screenshot

![Screenshot 2023-04-26 at 7 35 03 PM](https://user-images.githubusercontent.com/22123586/234745428-a0be7cf0-7c16-4c87-bf61-fa8e80f7cb69.png)

**Comment by awaik** [comment]
Please check the data types, I will add it to the docs.

<img width="1006" alt="Screenshot 2023-04-27 at 09 56 32" src="https://user-images.githubusercontent.com/5330807/234784223-bea15356-e152-4d14-9311-3a3a6f00086f.png">

**Comment by blacknight9** [comment]
> Please check the data types, I will add it to the docs.
> 
> <img alt="Screenshot 2023-04-27 at 09 56 32" width="1006" src="https://user-images.githubusercontent.com/5330807/234784223-bea15356-e152-4d14-9311-3a3a6f00086f.png">

exactly the same

**Comment by awaik** [comment]
Can you send me the code, I'll check it.
Please double-check that you have Maps - in the case of Array instead of map you will have this error.

**Comment by blacknight9** [comment]
> 


i copied your code and i was just testing

```
class Stories extends StatelessWidget {
   Stories({Key? key}) : super(key: key);

 static const String collectionDbName = 'stories';

final controller = Get.put(ProfilePageScreenController());

  final CollectionReference dbInstance =
  FirebaseFirestore.instance.collection(collectionDbName);

  @override
  Widget build(BuildContext context) {


    return FlutterInstagramStories(
      collectionDbName: collectionDbName,
      showTitleOnIcon:true,
      backFromStories: () {
        controller.backFromStoriesAlert(context);
      },
      iconTextStyle: const TextStyle(
        fontSize: 14.0,
        color: Colors.white,
      ),
      iconImageBorderRadius: BorderRadius.circular(15.0),
      iconBoxDecoration: const BoxDecoration(
        borderRadius: BorderRadius.all(Radius.circular(15.0)),
        color: Color(0xFFffffff),
        boxShadow: [
          BoxShadow(
            color: Color(0xff333333),
            blurRadius: 10.0,
            offset: Offset(
              0.0,
              4.0,
            ),
          ),
        ],
      ),
      iconWidth: 150.0,
      iconHeight: 150.0,
      textInIconPadding:
      const EdgeInsets.only(left: 8.0, right: 8.0, bottom: 12.0),
      //how long story lasts in seconds
      imageStoryDuration: 7,
      progressPosition: ProgressPosition.top,
      repeat: true,
      inline: false,
      languageCode: 'en',
      backgroundColorBetweenStories: Colors.black,
      closeButtonIcon: const Icon(
        Icons.close,
        color: Colors.white,
        size: 28.0,
      ),
      closeButtonBackgroundColor: const Color(0x11000000),
      sortingOrderDesc: true,
      lastIconHighlight: true,
      lastIconHighlightColor: myPrimaryAccent,
      lastIconHighlightRadius: const Radius.circular(15.0),
      captionTextStyle: const TextStyle(
        fontSize: 22,
        color: Colors.white,
      ),
      captionMargin: const EdgeInsets.only(
        bottom: 50,
      ),
      captionPadding: const EdgeInsets.symmetric(
        horizontal: 24,
        vertical: 8,
      ),
    );


  }
}

```

**Comment by awaik** [comment]
This is definetaly the data structure.
Please check maps, arrays one more time.
I've updated example - it works, so the reason is in the database.

**Comment by Ozgur-Abi** [comment]
Did you have any success solving the issue? I have the same database schema but although i can see previews, when i try to view stories I get this error

**Comment by awaik** [comment]
Please check arrays and maps as on the screenshots upwards.

**Comment by Ozgur-Abi** [comment]
> Please check arrays and maps as on the screenshots upwards.

I fixed it with this suggestion,
https://github.com/awaik/flutter_instagram_stories/issues/16
And created a fork of it https://github.com/Ozgur-Abi/flutter_instagram_stories_fixed.

However, video stories still don't work, on both your version and my fork. I tried it with both on my database and your database and example code. Video shows as loading but never loads.

**Comment by awaik** [comment]
Can you please make a video with database - when you point a mouse it displays field type.
In this case I will know for sure that the data structure is ok and help to fix it.

**Comment by danbeyene** [comment]
> > Please check arrays and maps as on the screenshots upwards.
> 
> I fixed it with this suggestion, #16 And created a fork of it https://github.com/Ozgur-Abi/flutter_instagram_stories_fixed.
> 
> However, video stories still don't work, on both your version and my fork. I tried it with both on my database and your database and example code. Video shows as loading but never loads.

uncomment setstate()  - inside story_video.dart - code snippet ---   playerController!.initialize().then((v) {
            setState(() {});
            widget.storyController!.play();
          });

**Comment by azappz** [comment]
the package not work, the story cover show but when i click on it u got null error, i tried all the solutions and didn't work
